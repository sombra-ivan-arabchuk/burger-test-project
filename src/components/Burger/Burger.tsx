import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import BurgerIngredient from '../BurgerIngredient/BurgerIngredient';

export interface BurgerIngredients {
  [key: string]: number;
}

export interface BurgerProps {
  ingredients: BurgerIngredients;
  name?: string;
  isEditing?: boolean;
}

const Burger: FunctionComponent<BurgerProps> = ({
  ingredients,
  name,
  isEditing = false,
}) => {
  const transformedIngredients = Object.keys(ingredients)
    .map((igKey: string) => {
      return [...Array(ingredients[igKey])].map((_, i) => (
        <BurgerIngredient key={igKey + i} type={igKey} />
      ));
    })
    .reduce((arr, el) => {
      return arr.concat(el);
    }, []);
  if (transformedIngredients.length === 0) {
    transformedIngredients.push(<p>please start adding ingredients</p>);
  }
  return (
    <BurgerContainer>
      {!isEditing && (
        <div
          style={{ fontSize: '20px', fontWeight: 'bold', marginBottom: '10px' }}
        >
          {name}
        </div>
      )}
      <BurgerIngredient type="bread-top" />
      {transformedIngredients}
      <BurgerIngredient type="bread-bottom" />
    </BurgerContainer>
  );
};

const BurgerContainer = styled.div`
  width: 100%;
  margin: auto;
  height: 100%;
  text-align: center;
  font-weight: bold;
  fontsize: 1.2rem;
`;

export default Burger;
