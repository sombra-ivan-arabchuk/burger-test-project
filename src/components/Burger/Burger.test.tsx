import React from 'react';
import {
  render,
  RenderResult,
  waitForElement,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Burger from './Burger';

export interface SetupProps {
  container: HTMLElement;
  utils: RenderResult;
}

jest.mock('../../hooks', () => ({ useAuth: (): object => ({}) }));

const data = { bacon: 2, cheese: 2, meat: 2, salad: 2 };

const setup = (): SetupProps => {
  // eslint-disable-next-line no-undef
  const utils = render(
    <Burger ingredients={data} name={'test burger'} isEditing={false} />,
  );
  return {
    container: utils.container,
    utils,
  };
};

describe('Catalog', () => {
  it('check for displaying burger name in not editing mode', async () => {
    const { utils } = setup();
    // I assume we will get this element because it is not editing mode
    const nameComponent = await waitForElement(() =>
      utils.findByText('test burger'),
    );
    expect(nameComponent).toHaveStyle('font-weight: bold');
  });
});
