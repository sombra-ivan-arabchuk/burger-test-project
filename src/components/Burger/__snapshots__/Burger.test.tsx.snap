// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders 1`] = `
<div
  class="sc-AykKG bQsbtM"
>
  <div
    style="font-size: 20px; font-weight: bold; margin-bottom: 10px;"
  />
  <div
    class="sc-AykKD bULVsK"
  >
    <div
      class="sc-AykKE fNTgGa"
    />
    <div
      class="sc-AykKF fMQbQa"
    />
  </div>
  <div
    style="width: 80%; height: 3%; margin: 2% auto;"
  />
  <div
    style="width: 80%; height: 3%; margin: 2% auto;"
  />
  <div
    style="width: 90%; height: 4.5%; margin: 2% auto; border-radius: 20px;"
  />
  <div
    style="width: 90%; height: 4.5%; margin: 2% auto; border-radius: 20px;"
  />
  <div
    style="width: 80%; height: 8%; margin: 2% auto; border-radius: 15px;"
  />
  <div
    style="width: 80%; height: 8%; margin: 2% auto; border-radius: 15px;"
  />
  <div
    style="width: 85%; height: 7%; margin: 2% auto; border-radius: 20px;"
  />
  <div
    style="width: 85%; height: 7%; margin: 2% auto; border-radius: 20px;"
  />
  <div
    class="sc-AykKC hTTtHC"
  />
</div>
`;
